CenterForOpenScience/osf.io
# [Feature] Replace zeroclipboardjs with clipboardjs [OSF-7771]

## Purpose

Replace the current module (zeroclipboard) used for copy-to-clipboard buttons
with clipboard which does not rely on Flash. Removing the flash requirement allows copying
to clipboard to work effortlessly in modern browsers.

## Changes
- Adds clipboard as bower dependency
- Removes zeroclipboard as bower dependency
- Modifies the clipboard abstract file to use clipboard
- Modifies AddonNodeConfig and CitationsNodeConfig tests to use clipboard

## QA Considerations
- This fix affects all the places on the osf that have copy-to-clipboard button. So far, I know:
files view page under **revisions**, files views page under **share** (must be public project), and view only links table in **Contributors**.

- In all the above mentioned places: please make sure the tooltip 'Copied' shows upon hitting the copy button.

## Ticket
https://openscience.atlassian.net/browse/OSF-7771
